### PR TITLE
[release/9.0] Stable branding for GA

### PR DESCRIPTION
**Pull request by carlossanlop**

### CODE REVIEW

**Comment by carlossanlop** [comment]
The emsdk stable branding deps flow PR is https://github.com/dotnet/runtime/pull/108898

**Review by ViktorHofer — APPROVED** [review]

**Comment by mmitche** [comment]
@carlossanlop @akoeplinger Test is failing on check a stable branding string. emsdk is merged but I don't see any failrues here that are a result of not flowing emsdk bits yet. Test failure is here:

```
#if STABILIZE_PACKAGE_VERSION
            // a stabilized version looks like 8.0.0
            Assert.DoesNotContain("-", version);
            Assert.True(Version.TryParse(version, out Version _));
#else
            // a non-stabilized version looks like 8.0.0-preview.5.23280.8 or 8.0.0-dev
            Assert.Contains("-", version);
            var versionNumber = version.Substring(0, version.IndexOf("-"));
            Assert.True(Version.TryParse(versionNumber, out Version _));
#endif
```

**Comment by carlossanlop** [comment]
The test is expecting to _not_ see the `"-"` character in the version, but PR CI results are (AFAIK) always expected to have the `"-ci"` suffix added to them:

```
[19:03:40] info: [FAIL] System.Runtime.InteropServices.RuntimeInformationTests.DescriptionNameTests.VerifyFrameworkDescriptionContainsCorrectVersion
[19:03:40] info: Assert.DoesNotContain() Failure: Sub-string found
[19:03:40] info:               ↓ (pos 5)
[19:03:40] info: String: "9.0.0-ci"
[19:03:40] info: Found:  "-"
[19:03:40] info:    at System.Runtime.InteropServices.RuntimeInformationTests.DescriptionNameTests.VerifyFrameworkDescriptionContainsCorrectVersion()
[19:03:40] info:    at System.Reflection.MethodBaseInvoker.InterpretedInvoke_Method(Object obj, IntPtr* args)
[19:03:40] info:    at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
[19:03:40] info: Finished:    System.Runtime.InteropServices.RuntimeInformation.Tests.dll
```

I suspect the test will pass in rolling builds where `"-ci"` is not added.

**Comment by mmitche** [comment]
> The test is expecting to _not_ see the `"-"` character in the version, but PR CI results are (AFAIK) always expected to have the `"-ci"` suffix added to them, and local builds are expected to have the `"-dev"` suffix.
> 
> ```
> [19:03:40] info: [FAIL] System.Runtime.InteropServices.RuntimeInformationTests.DescriptionNameTests.VerifyFrameworkDescriptionContainsCorrectVersion
> [19:03:40] info: Assert.DoesNotContain() Failure: Sub-string found
> [19:03:40] info:               ↓ (pos 5)
> [19:03:40] info: String: "9.0.0-ci"
> [19:03:40] info: Found:  "-"
> [19:03:40] info:    at System.Runtime.InteropServices.RuntimeInformationTests.DescriptionNameTests.VerifyFrameworkDescriptionContainsCorrectVersion()
> [19:03:40] info:    at System.Reflection.MethodBaseInvoker.InterpretedInvoke_Method(Object obj, IntPtr* args)
> [19:03:40] info:    at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
> [19:03:40] info: Finished:    System.Runtime.InteropServices.RuntimeInformation.Tests.dll
> ```
> 
> I suspect the test will pass in rolling builds where `"-ci"` or `"-dev"` is not added.
> 
> Fixing the test by making it more specific.

Hmm, the shipping assets maybe should not have -ci. At least, when i build winforms stable with -ci, shipping packages are 9.0.0. Maybe this is just specific to the framework description? We def need to verify on an official build.

**Review by dkurepa — APPROVED** [review]

**Comment by dkurepa** [comment]
We didn't see any `-ci` suffix in 9.0.0 test build https://dev.azure.com/dnceng/internal/_build/results?buildId=2531164&view=results. So I guess it's just a PR build?

File: src/libraries/System.Runtime/tests/System.Runtime.InteropServices.RuntimeInformation.Tests/DescriptionNameTests.cs [hunk 1]
```diff
@@ -198,11 +198,21 @@ public void VerifyFrameworkDescriptionContainsCorrectVersion()
             Assert.DoesNotContain("+", version); // no git hash
 
 #if STABILIZE_PACKAGE_VERSION
-            // a stabilized version looks like 8.0.0
-            Assert.DoesNotContain("-", version);
-            Assert.True(Version.TryParse(version, out Version _));
+            // A stabilized version looks like 9.0.0
+            // In local builds, the version looks like 9.0.0-dev, and in PRs CI, the version looks like 9.0.0-ci.
+            // Both are acceptable.
+            if (!version.Contains("-dev") && !version.Contains("-ci"))
```
Comment (akoeplinger) [inline src/libraries/System.Runtime/tests/System.Runtime.InteropServices.RuntimeInformation.Tests/DescriptionNameTests.cs:204]: This is not correct. The FrameworkDescription should always contain the stabilized version irrespective of the build environment.

This is a regression caused by https://github.com/dotnet/runtime/pull/102164. Before it was reading the AssemblyInformationalVersion from System.Private.CoreLib.dll which is `9.0.0`, now it's reading it from the generators assembly and that contains `9.0.0-dev` (basically what Stephen noticed in https://github.com/dotnet/runtime/pull/102164#discussion_r1601772016)

**Review by akoeplinger — CHANGES_REQUESTED** [review]

File: src/libraries/System.Runtime/tests/System.Runtime.InteropServices.RuntimeInformation.Tests/DescriptionNameTests.cs (hunk 1 above)
Comment (akoeplinger) [reply src/libraries/System.Runtime/tests/System.Runtime.InteropServices.RuntimeInformation.Tests/DescriptionNameTests.cs]: I pushed a fix. There is potential to clean up overriding the version in SPC.dll now I think but that can be done separately.

**Review by akoeplinger — APPROVED** [review]

**Comment by ViktorHofer** [comment]
Should these lines now get removed (at least in main when we forward port this change)?

- https://github.com/dotnet/runtime/blob/ee99d7c50bed644bbb53c30b73cd21c7fe6c9fc2/src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj#L47-L49
- https://github.com/dotnet/runtime/blob/ee99d7c50bed644bbb53c30b73cd21c7fe6c9fc2/src/mono/System.Private.CoreLib/System.Private.CoreLib.csproj#L34-L36
- https://github.com/dotnet/runtime/blob/ee99d7c50bed644bbb53c30b73cd21c7fe6c9fc2/src/coreclr/nativeaot/Directory.Build.props#L28-L30

**Comment by akoeplinger** [comment]
Yeah I mentioned it above.

**Comment by mmitche** [comment]
Looks like some additional test failures that may be test issues rather than product issues

**Comment by carlossanlop** [comment]
Build Analysis says most failures are known. There is one in Installer Host Activation that I have not seen before. @NikolaMilosavljevic does this look familiar? https://dev.azure.com/dnceng-public/public/_build/results?buildId=844459&view=logs&j=34fcf1e9-20e8-52c9-c21d-1ed3608ac124&t=3f5386ba-6526-5d7f-4342-42281a824185&l=489

**Comment by mmitche** [comment]
image

Those are the ones i think are interesting.

**Comment by carlossanlop** [comment]
> Those are the ones i think are interesting.

The failure is coming from the build output argument passed to this method: https://github.com/dotnet/runtime/blob/9305d7f71d73c1d1edeb2a06478c998e40deda8d/src/mono/wasi/Wasi.Build.Tests/BuildTestBase.cs#L345-L355

**Comment by carlossanlop** [comment]
We generate the build out of this portion of the code, which we then pass to AssertRuntimePackPath. I wonder if we need to pass an environment variable to indicate that we are using a stabilizing build.

https://github.com/dotnet/runtime/blob/9305d7f71d73c1d1edeb2a06478c998e40deda8d/src/mono/wasi/Wasi.Build.Tests/BuildTestBase.cs#L243-L253

**Comment by carlossanlop** [comment]
Good news: The release/9.0 rolling build that included this change did _not_ hit the failures reported above: https://dev.azure.com/dnceng/internal/_build/results?buildId=2562281&view=results

So they indeed are only happening in PRs.

**Pull request merged**